src/Parsers/FunctionSecretArgumentsFinder.h

```diff
@@ -23,6 +55,485 @@ class FunctionSecretArgumentsFinder
             return count != 0 || !nested_maps.empty();
         }
     };
+
+    explicit FunctionSecretArgumentsFinder(std::unique_ptr<AbstractFunction> && function_) : function(std::move(function_)) {}
+
+    FunctionSecretArgumentsFinder::Result getResult() const { return result; }
+
+protected:
+    const std::unique_ptr<AbstractFunction> function;
+    Result result;
+
+    void markSecretArgument(size_t index, bool argument_is_named = false)
```

Comment:
Hmm, why is all of the implementation below in a header file?

Reply:
well, because it was taken out of previously existing header files... didn't give much thought to this issue...